### ISSUE TITLE
Certificate fields

### ISSUE THREAD

**Issue by ericlaw1979** [vcsjones/FiddlerCertGen#3]
`var cert = _generator.GenerateCertificate(GetRootCertificate(), GetEEPrivateKey(), new X500DistinguishedName("CN=FIDDLER_EE_DO_NOT_TRUST,OU=vcsjones.com"), new[] { sHostname });`

Am I correct in understanding that your certificates _require_ respect of the SubjectAltName since the SubjectCN will never match? Is it legal to have an IPAddress in a SubjectAltName field?

It may make sense for your generator to emulate the `CN/O/OU` of the existing certificate makers (ie `CN=DO_NOT_TRUST_FiddlerRoot, O=DO_NOT_TRUST, OU=Created by http://www.fiddler2.com`) because otherwise the "Remove Interception Certificates" button won't work as expected if the user uses your generator and then switches to another one without clicking the "Remove" button first. (This is something of a design problem in Fiddler itself).

**Comment by vcsjones** [comment]
> Am I correct in understanding that your certificates require respect of the SubjectAltName since the SubjectCN will never match?

Yes. Issue #1's goal is to reuse certificates to reduce the number of certificates by using wildcards.

Is it unrealistic to expect that most, if not all, UA's respect SANs?

> because otherwise the "Remove Interception Certificates" button won't work as expected if the user uses your generator and then switches to another one

The certificates are never put into the certificate store, other than the root. Is that still applicable?

**Comment by ericlaw1979** [comment]
> most, if not all, UA's respect SANs?

It's a great question -- I suspect that it's _probably_ true given CloudFlare, etc. But I think there probably are some niche clients that only look at CN. 

I'm additionally curious whether a literal IP address is legal in a SAN.

I'm also a bit worried that by putting your domain name in the OU, you might be taking on some attention you don't want, especially if some badware just picks up your generator and uses it.

> The certificates are never put into the certificate store, other than the root. 

Cleanup for the root does seem desirable, yes.

**Comment by vcsjones** [comment]
> I'm additionally curious whether a literal IP address is legal in a SAN.

SubjAltName does officially have a means of putting in an IP address, but last time I tested it (2012?) no browser supports anything in a SAN other than DNS Name.

`CERT_ALT_NAME` supports it when using `CERT_ALT_NAME_IP_ADDRESS` as the `choiceType`, but this project doesn't support it because it is of little value at the moment.

> I'm also a bit worried that by putting your domain name in the OU, you might be taking on some attention you don't want.

Agree! I chose my own domain so that if this code where used stupidly, then I wouldn't be directing traffic / support to you for something that some idiot like my self developed.
